apps/studio/components/interfaces/Storage/StorageExplorer/MoveItemsModal.tsx

```diff
@@ -1,90 +1,127 @@
-import { noop } from 'lodash'
-import { useEffect, useState } from 'react'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { type SubmitHandler, useForm } from 'react-hook-form'
+import { z } from 'zod'
 
-import { Button, Input, Modal } from 'ui'
-import { StorageItemWithColumn } from '../Storage.types'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogSection,
+  DialogSectionSeparator,
+  DialogTitle,
+  Form_Shadcn_,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Input_Shadcn_,
+} from 'ui'
+import { useStorageExplorerStateSnapshot } from 'state/storage-explorer'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
 
-interface MoveItemsModalProps {
-  bucketName: string
-  visible: boolean
-  selectedItemsToMove: StorageItemWithColumn[]
-  onSelectCancel: () => void
-  onSelectMove: (path: string) => void
+export interface MoveItemsModalProps {
+  onClose?: () => void
 }
 
-const MoveItemsModal = ({
-  bucketName = '',
-  visible = false,
-  selectedItemsToMove = [],
-  onSelectCancel = noop,
-  onSelectMove = noop,
-}: MoveItemsModalProps) => {
-  const [moving, setMoving] = useState(false)
-  const [newPath, setNewPath] = useState('')
+const PathSchema = z.object({
+  path: z.string().min(0, 'Please provide a valid path name'),
+})
```

Comment:
we probably want this to validate against a regex or something similar, so that only valid directory paths can be entered